pkg/uefi/file.go

```diff
@@ -173,13 +174,34 @@ const (
 	FileStateValid FileState = FileStateHeaderConstruction | FileStateHeaderValid | FileStateDataValid
 )
 
+type ThreeUint8 [3]uint8
+
+func (t *ThreeUint8) UnmarshalJSON(b []byte) error {
+	if len(b) == 0 {
+		return fmt.Errorf("Cannot unmarshal 3 uint8 from %v\n", b)
+	}
+	t[0] = b[0]
+	if len(b) > 1 {
+		t[1] = b[1]
+	}
+	if len(b) > 2 {
+		t[2] = b[2]
+	}
+	return nil
+}
+
+func (t *ThreeUint8) MarshalJSON() ([]byte, error) {
+	res := uint32(t[0]) + uint32(t[1])<<8 + uint32(t[2])<<16
```

Comment:
Can you use Read3Size? There's already a helper function that does this.

Reply:
now that you showed me how, yea, thank you :)